imageio/plugins/_freeimage.py

```diff
@@ -953,28 +953,20 @@ def set_image_data(self, array):
             n_channels = shape[0]
         
         def n(arr):  # normalise to freeimage's in-memory format
-            return arr.T[:, ::-1]
+            return arr[::-1].T
         wrapped_array = self._wrap_bitmap_bits_in_array(w_shape, dtype, True)
         # swizzle the color components and flip the scanlines to go to
         # FreeImage's BGR[A] and upside-down internal memory format
-        if len(shape) == 3:
+        if len(shape) == 3 and isle and dtype.type == numpy.uint8:
             R = array[:, :, 0]
             G = array[:, :, 1]
             B = array[:, :, 2]
-        
-            if isle:
-                if dtype.type == numpy.uint8:
-                    wrapped_array[0] = n(B)
-                    wrapped_array[1] = n(G)
-                    wrapped_array[2] = n(R)
-                elif dtype.type == numpy.uint16:
-                    wrapped_array[0] = n(R)
-                    wrapped_array[1] = n(G)
-                    wrapped_array[2] = n(B)
-                #
-                if shape[2] == 4:
-                    A = array[:, :, 3]
-                    wrapped_array[3] = n(A)
+            wrapped_array[0] = n(B)
+            wrapped_array[1] = n(G)
+            wrapped_array[2] = n(R)
+            if shape[2] == 4:
+                A = array[:, :, 3]
+                wrapped_array[3] = n(A)
```

Comment:
Well, no.
isle says whether the system is big endian right? Which I guess rarely happens anymore. So I don't care much about that.
The problem with the old code was that if the image was a color image (len(shape) == 3) with a float32 dtype nothing would be written to wrapped_array.